src/odo.ts

```diff
@@ -367,8 +380,20 @@ export class OdoImpl implements Odo {
             // show no apps if output is not correct json
             // see https://github.com/openshift/odo/issues/1521
         }
-        const apps: string[] = data.map((value) => value.metadata.name);
-        return apps.map<OpenShiftObject>((value) => new OpenShiftObjectImpl(application, value, ContextType.COMPONENT, this, TreeItemCollapsibleState.Collapsed));
+        const componentObject = data.map(value => ({ name: value.metadata.name, source: value.spec.source }));
+
+        return componentObject.map<OpenShiftObject>((value) =>
+            {
+                let compSource: string = '';
+                if (value.source.startsWith('https://')) {
+                    compSource = 'git';
+                } else if (Archive.endsWithAny([".tar", ".jar", ".iso", ".tar.gz", ".zip"], value.source)) {
```

Comment:
We cannot just restrict to files with extensions for archives. There are many others out there: war, ear and etc.  It should check if ```value.source``` is path to the existing file.